.changeset/smooth-goats-agree.md

```diff
@@ -0,0 +1,49 @@
+---
+'astro': minor
+---
+
+Add support for generating multiple widths when using the Image component and a Picture component for supporting multiple formats.
+
+## `srcset` support
+
+Two new properties have been added to `Image` and `getImage`: `densities` and `widths`.
+
+These props can be used to generate a `srcset` attribute with multiple sources. For example:
```

Comment:
The `width / 2` is actually required! Because densities should never upscale, so your base image needs to be smaller than the densities it generates. (You can think of `width` as controlling the base image, whereas `densities` control its variants)

If you have an image that is originally 1280x720, and you ask for density 2x (2560x1440) without specifying a smaller size for the base image, Astro would have to upscale the original image to reach that size, since we never upscale, `density={['2x']}` would just get ignored.

Reply:
I kind of know I knew that at one point... and I agree, no need to get into that here. But it doesn't seem intuitive to me that an array with a single value of 2 produces both a 2x and a 3x in `srcset`. Admittedly, if I understood this better, I'd be better able to say what should/shouldn't go in a description here. Naively, I would have expected to need to specify two values in my array to get two generated.

Reply:
Oh, it doesn't sorry!

If you do `densities={[2]}`, it'll only have the original density (in the `src` attribute) and the 2x in `srcset`.